adagrams/game.py

```diff
@@ -1,11 +1,64 @@
+from random import sample
 def draw_letters():
-    pass
+    hand = sample(['A','B','C','D','E','F','G','H','I','J','K','L',\
+                   'M','N','O','P','Q','R','S','T','U','V','W','X',\
+                   'Y','Z'], counts = [9,2,2,4,12,2,3,2,9,1,1,4,2,6,\
+                                       8,2,1,6,4,6,4,2,2,1,2,1], k = 10)
+    return hand
 
 def uses_available_letters(word, letter_bank):
-    pass
+    letter_bank_dict = {}
+    for letter in letter_bank:
+        if letter not in letter_bank_dict:
+            letter_bank_dict[letter] = 1
+        else:
+            letter_bank_dict[letter] += 1
+    for letter in word.upper():
+        if letter not in letter_bank_dict:
+            return False
+        elif letter_bank_dict[letter] == 0:
+            return False
+        else:
+            letter_bank_dict[letter] -= 1
+    return True
+
 
 def score_word(word):
-    pass
+    scores = {
+        "AEIOULNRST" : 1,
+        "DG" : 2,
+        "BCMP" : 3,
+        "FHVWY" : 4,
+        "K" : 5,
+        "JX" : 8,
+        "QZ" : 10
+    }
+
+    score = [val for letter in word.upper() for key, val in scores.items() if letter in key]
+
+    if len(score) >= 7:
+        score.append(8)
+
+    return sum(score)
 
 def get_highest_word_score(word_list):
```

Comment:
Just wanted to add some commentary to this get_highest_word_score() func! I initially considered utilizing a word:score dictionary here, but settled on the tuples because of ease and clarity in accessing/sorting each word. I realize my solution might not be as efficient as a dict that only goes through one score sort and one tie breaker sort, however, and I've played around with a few varying implementations since submission to try and maximize efficiency thereby!